### PR TITLE
Nv 241 enable runtime dynamic environment

### PR DESCRIPTION
**Pull request by davidsoderberg**

### CODE REVIEW

File: .github/workflows/dev-deploy-web.yml
Comment (scopsy) [inline .github/workflows/dev-deploy-web.yml:65]: Is it possible to avoid the `cat` here? It might expose some secrets to the actions logs.

File: .github/workflows/dev-deploy-web.yml
Comment (scopsy) [inline .github/workflows/dev-deploy-web.yml:69]: Great!

File: .github/workflows/prod-deploy-web.yml
Comment (scopsy) [inline .github/workflows/prod-deploy-web.yml:64]: If we can remove this one aswell

File: apps/web/Dockerfile
Comment (scopsy) [inline apps/web/Dockerfile:29]: Can we only `build:web` here? because it's not necessary to build the entire monorepo here.

File: apps/web/package.json
Comment (scopsy) [inline apps/web/package.json:98]:
```suggestion
    "start": "npm run envsetup && PORT=4200 react-app-rewired start",
```

File: apps/web/package.json
Comment (scopsy) [inline apps/web/package.json:102]: I'm a bit concerned with windows users here, there is no problem that this will run inside the Dockerfile in which control the OS. But from the looks of it, windows users won't be able to start the application locally. 

Is there another way to handle this locally?

File: apps/web/src/config/index.ts
Comment (scopsy) [inline apps/web/src/config/index.ts:14]: Love the new folder and config file to handle everything from here. Love that!

**Review by scopsy — COMMENTED** [review]
@davidsoderberg This looks great! Loved the new config file. A couple of comments and my main concern is for windows users running the project locally. Let me know what you think

File: apps/web/package.json
Comment (davidsoderberg) [reply apps/web/package.json:102]: I guess it is possible to create an env-config.js in public folder and use it for windows users...?

File: .github/workflows/dev-deploy-web.yml
Comment (davidsoderberg) [reply .github/workflows/dev-deploy-web.yml:65]: Should not be a problem

File: .github/workflows/prod-deploy-web.yml
Comment (davidsoderberg) [reply .github/workflows/prod-deploy-web.yml:64]: Should not be a problem

File: apps/web/package.json
Comment (scopsy) [reply apps/web/package.json:102]: I'm thinking, if we can skip this altogether and only rely on fallback values in the config.ts file you created. Because today, you can run the project without process.env files altogether.

In this way we can only run envsetup in docker and in github actions where it's needed. And locally both MAC and Windows users won't need to run this command.

So I think, you are correct with created a default file there, and we can just unify the mac and windows experience.

Comment (davidsoderberg) [reply apps/web/package.json:102]: Hm having hard time to follow you in your comment... 
So envsetup for docker and github actions should be easy :)
So we should add a default env-config.js file?

Comment (scopsy) [reply apps/web/package.json:102]: Yes just like you suggested. 

And locally never run the envsetup command. So that way mac and windows users will have the same experience.

Comment (davidsoderberg) [reply apps/web/package.json:102]: hm I guess thats fine because those envs are public anyway...

**Review by scopsy — APPROVED** [review]
Let's retest this in DEV environment and than we can merge 🎉

**Comment by davidsoderberg** [comment]
this works in dev now 😄

**Comment by scopsy** [comment]
@davidsoderberg Great! @ainouzgali pushed a rename of the SDK package. Could you just resolve the conflicts and I think we can merge :)

File: .github/workflows/dev-deploy-web.yml
Comment (scopsy) [inline .github/workflows/dev-deploy-web.yml:61]:
```suggestion
          echo REACT_APP_WIDGET_EMBED_PATH="https://dev.sdk.novu.co/sdk.umd.min.js" >> .env
```
@davidsoderberg 
After the merge this will be, if you could just recheck all the places we used "SDK" and SDK, to become "embed"

**Comment by scopsy** [comment]
@davidsoderberg Do we need to perform more changes here? Or we can push this?

**Comment by davidsoderberg** [comment]
We can push this :)

**Pull request merged**
